getter.go

```diff
@@ -152,26 +152,30 @@ func (g *getter) initChunks() {
 
 func (g *getter) worker() {
 	for c := range g.getCh {
-		g.retryGetChunk(c)
+		if err := g.retryGetChunk(c); err != nil {
+			close(g.readCh)
+			break
+		}
 	}
 }
 
-func (g *getter) retryGetChunk(c *chunk) {
+func (g *getter) retryGetChunk(c *chunk) error {
 	var err error
 	c.b = <-g.sp.get
 	for i := 0; i < g.c.NTry; i++ {
 		err = g.getChunk(c)
 		if err == nil {
-			return
+			return nil
 		}
 		logger.debugPrintf("error on attempt %d: retrying chunk: %v, error: %s", i, c.id, err)
 		time.Sleep(time.Duration(math.Exp2(float64(i))) * 100 * time.Millisecond) // exponential back-off
 	}
 	select {
 	case <-g.quit: // check for closed quit channel before setting error
-		return
+		return nil
 	default:
 		g.err = err
+		return err
```

Comment:
Could we rely on reading if `g.err` gets set instead of closing in the reader? Generally you'd only want to close a channel if you're the writer.

Reply:
I had considered that, but I think checking `g.err` would require taking a lock (though maybe that's not so bad). For what its worth, I understood the `worker()` goroutine to be the [writer](https://github.com/github/s3gof3r/blob/master/getter.go#L205) for the `g.readCh` channel. But the panic issue you raised above may rule out that approach anyway.